dj/api/nodes.py

```diff
@@ -259,19 +261,16 @@ def delete_a_node(name: str, *, session: Session = Depends(get_session)):
         downstream.current.status = NodeStatus.INVALID
         session.add(downstream)
 
-    # If the node is a dimension, find all columns that
```

Comment:
Hmm... don't we still need to do a version of this where we invalidate the downstream nodes? And then for the dimension links, we'd need to invalidate those as well I think. That might need to be stored at the attribute level, not sure.

Reply:
I was thinking about how to handle this and I can't come up with anything that doesn't have some tradeoffs. If deactivating a dimension invalidates everything linked to it, that would actually have a larger blast radius than we need. Take the example of a valid metric node that has access to 5 dimension nodes and all of the upstream nodes are valid. Just because one of those dimension nodes are deactivated, I don't think we should mark the metric node as invalid. It's still largely valid in that you could use it with any of the 4 dimension nodes left and build a cube with other metrics that share one or more dimensions.

On the other hand, we don't want to show a deactivated dimension node anywhere since they shouldn't be allowed for use in building a SQL query or retrieving data. What if we left the nodes using a deactivated dimension as valid but made sure to not display that dimension node in any listings and also made sure to not allow using it to get SQL/data?

Reply:
> Just because one of those dimension nodes are deactivated, I don't think we should mark the metric node as invalid.

Yeah, agreed that I don't think the node should be marked as invalid, but the dimension link needs to be marked as invalid (or hidden in the ways you describe). While we could just filter out this invalid dimension link when we return dimensions for metrics, we still store the link on the column attributes -- should this also have a valid/invalid setting to indicate a deactivated vs activated dimension node?

Reply:
I think it may be helpful to think about this space in the following way:
- data flow graph (DAG): 
   parent node deactivation --> all child nodes become invalid
- dimension attribute graph (not a DAG) 
   dimension deactivation --> children nodes are still valid, just the dimension/column attribute becomes invalid --> should affect only the functionality that depends on the dimension attributes, nothing else.

@shangyian @samredai would you agree? Sorry if this is obvious but I wanted to spell it out.

Reply:
@agorajek That separation makes sense to me! Definitely good to spell it out

Reply:
@agorajek agreed! I'll work on adding a user facing description of these behaviors to the docs as well.